rotkehlchen/chain/evm/decoding/interfaces.py

```diff
@@ -202,43 +198,32 @@ def __init__(
             cache_type_to_check_for_freshness: CacheType,
             query_data_method: Union[
                 Callable[['OptimismInquirer'], Optional[list['VelodromePoolData']]],
-                Callable[['EthereumInquirer'], Optional[list['CurvePoolData']]],
-            ],
-            save_data_to_cache_method: Union[
-                Callable[['DBCursor', 'DBHandler', list['CurvePoolData']], None],
-                Callable[['DBCursor', 'DBHandler', list['VelodromePoolData']], None],
-            ],
-            read_pools_and_gauges_from_cache_method: Union[
-                Callable[[], tuple[set[ChecksumEvmAddress], set[ChecksumEvmAddress]]],
-                Callable[[], 'READ_CURVE_DATA_TYPE'],
+                Callable[['EthereumInquirer'], Optional[list]],
             ],
+            save_data_to_cache_method: Callable[['DBCursor', 'DBHandler', list], None],
+            read_data_from_cache_method: Callable[[], tuple[Union[dict[ChecksumEvmAddress, Any], set[ChecksumEvmAddress]], ...]],  # noqa: E501
```

Comment:
Guess same question I had before. The reason for `tuple[XXX, ...]` is to accomodate the difference between curve/velodrome having pools and gauges and convex having only pool? 

I would mention it here. Or probably best in the docstring below. At like 211. Explain this detail and why you have it like this.

Reply:
I explained in line 211